### PR TITLE
Update tc39 tests

### PR DESCRIPTION
**Pull request by MStoykov**
This will end up being squashed in 1 commit it mostly exists as many so that `breaking_test_errors.json` can be seen changing

Here is a [list](https://gist.github.com/MStoykov/0ba38c3f23b7b095c631e5c121a973cb) of all the "new" Skipped tests. New here is defined as from after I updated tc39 (first commit) to the latest one where I disable a bunch of the ones that have no chance to run. 
If you see a test about something you have no idea what it is ... it likely is still a proposal ;)

### CODE REVIEW

**Review by oleiade — APPROVED** [review]
LGTM 👍🏻

File: js/tc39/tc39_test.go [hunk 1]
```diff
@@ -51,6 +51,45 @@ var (
...
+		"numeric-separator-literal",
+		"Object.fromEntries",
+		"Object.hasOwn",
+		"__getter__",
+		"__setter__",
+		"ShadowRealm",
+		"SharedArrayBuffer",
+		"error-cause",
+		"resizable-arraybuffer", // stage 3 as of 2021 https://github.com/tc39/proposal-resizablearraybuffer
+
```
Comment (codebien) [inline js/tc39/tc39_test.go:88]: Out of curiosity, is this newline relevant for separating something? If yes, I would add a comment as you did before for `goja`.

**Review by codebien — COMMENTED** [review]
LGTM, just a comment.

File: js/tc39/tc39_test.go (hunk 1 above)
Comment (MStoykov) [reply js/tc39/tc39_test.go:88]: just to separate them, the above list is from goja, while the below is stuff I added (goja doesn't run those through different means)

**Review by codebien — APPROVED** [review]

**Pull request merged**